### PR TITLE
Improve Message Coding table headers 

### LINKED ISSUE #120: Improve Message Coding table headers
We're seeing people interpret the dashboard as showing the total number of messages received in each dataset, and the fraction of WS/NC being interpreted as representative of the analysis dataset. These interpretations, however, are incorrect because the dashboard actually shows the number of unique messages and the fraction of those unique messages.

We could start to resolve this by renaming the table headers to e.g. "Unique Texts" rather than "Messages" etc. We might also add a description underneath the table that explains those columns and what can and cannot be inferred from them.

### PR DESCRIPTION
**Pull request by DanielMwendwa**
closes #120

### CODE REVIEW

File: public/UI.js [hunk 1]
```diff
@@ -39,7 +39,64 @@ class UIController {
...
+                                        <th scope="row">Wrong Scheme messages</th>
+                                        <td>shows the number of ‘messages with a label’ that contain a checked “WS (Wrong Scheme)” label.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">WS %</th>
+                                        <td></td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Not Coded messages</th>
+                                        <td>shows the number of messages with a label’ that contain a checked “NC (Not Coded)” label.</td>
```
Comment (IsaackMwenda) [inline public/UI.js]:
```suggestion
                                        <td>Shows the number of messages with a label’ that contain a checked “NC (Not Coded)” label.</td>
```

File: public/UI.js [hunk 2]
```diff
@@ -39,7 +39,64 @@ class UIController {
...
+                                        <th scope="row">Unique Texts with a label</th>
+                                        <td>shows the number of unique texts which contain a checked (i.e. manually verified) label in at least one code scheme</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Done</th>
+                                        <td></td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Wrong Scheme messages</th>
+                                        <td>shows the number of ‘messages with a label’ that contain a checked “WS (Wrong Scheme)” label.</td>
```
Comment (IsaackMwenda) [inline public/UI.js]:
```suggestion
                                        <td>Shows the number of ‘messages with a label’ that contain a checked “WS (Wrong Scheme)” label.</td>
```

File: public/UI.js [hunk 3]
```diff
@@ -39,7 +39,64 @@ class UIController {
...
+                                        <th scope="row">Dataset</th>
+                                        <td></td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Unique Texts</th>
+                                        <td>  shows the total number of unique texts in the dataset. Note that to reduce labelling load, Coda only displays each unique message text once, so this is not the total number of messages received.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Unique Texts with a label</th>
+                                        <td>shows the number of unique texts which contain a checked (i.e. manually verified) label in at least one code scheme</td>
```
Comment (IsaackMwenda) [inline public/UI.js]:
```suggestion
                                        <td>Shows the number of unique texts which contain a checked (i.e. manually verified) label in at least one code scheme</td>
```

File: public/UI.js [hunk 4]
```diff
@@ -39,7 +39,64 @@ class UIController {
...
+                                    </tr>
+                                </thead>
+                                <tbody>
+                                    <tr>
+                                        <th scope="row">Dataset</th>
+                                        <td></td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Unique Texts</th>
+                                        <td>  shows the total number of unique texts in the dataset. Note that to reduce labelling load, Coda only displays each unique message text once, so this is not the total number of messages received.</td>
```
Comment (IsaackMwenda) [inline public/UI.js]:
```suggestion
                                        <td>Shows the total number of unique texts in the dataset. Note that to reduce labelling load, Coda only displays each unique message text once, so this is not the total number of messages received.</td>
```

File: public/UI.js [hunk 5]
```diff
@@ -39,7 +39,64 @@ class UIController {
...
                     <tbody id="coding-status-body"></tbody>
                 </table>
-            <div id="last-update">Last updated: </div>
+                <div id="last-update">Last updated: </div>
+                <div class="accordion" id="accordionExample">
+                    <div class="card">
+                        <div class="card-heade" id="headingOne">
+                            <h2 class="mb-0">
+                                <button class="btn btn-brown" type="button" data-toggle="collapse" data-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
+                                Message Coding table headers Definitions
```
Comment (IsaackMwenda) [inline public/UI.js]:
```suggestion
                                Message Coding Table Header Definitions
```

File: public/UI.js [hunk 6]
```diff
@@ -39,7 +39,64 @@ class UIController {
...
+                                        <th scope="row">Unique Texts</th>
+                                        <td>  shows the total number of unique texts in the dataset. Note that to reduce labelling load, Coda only displays each unique message text once, so this is not the total number of messages received.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Unique Texts with a label</th>
+                                        <td>shows the number of unique texts which contain a checked (i.e. manually verified) label in at least one code scheme</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Done</th>
+                                        <td></td>
```
Comment (IsaackMwenda) [inline public/UI.js]: % of labeled unique texts over total unique texts.

File: public/UI.js [hunk 7]
```diff
@@ -39,7 +39,64 @@ class UIController {
...
+                <div class="accordion" id="accordionExample">
+                    <div class="card">
+                        <div class="card-heade" id="headingOne">
+                            <h2 class="mb-0">
+                                <button class="btn btn-brown" type="button" data-toggle="collapse" data-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
+                                Message Coding table headers Definitions
+                                </button>
+                            </h2>
+                        </div>
+                        <div id="collapseOne" class="collapse show" aria-labelledby="headingOne" data-parent="#accordionExample">
```
Comment (IsaackMwenda) [inline public/UI.js]:
```suggestion
                        <div id="collapseOne" class="collapse" aria-labelledby="headingOne" data-parent="#accordionExample">
```

Comment (IsaackMwenda) [inline public/UI.js]: It would be better if it scrolls up once someone clicks the button.

**Review by IsaackMwenda — CHANGES_REQUESTED** [review]

File: public/UI.js [hunk 8]
```diff
@@ -42,7 +42,64 @@ class UIController {
...
+                            <table class="table table-bordered table-hover shadow">
+                                <thead>
+                                    <tr class="table-heading">
+                                        <th scope="col">Table Headers</th>
+                                        <th scope="col">Description</th>
+                                    </tr>
+                                </thead>
+                                <tbody>
+                                    <tr>
+                                        <th scope="row">Dataset</th>
```
Comment (IsaackMwenda) [inline public/UI.js:69]: The collection of messages for a given radio episode or survey

File: public/UI.js [hunk 9]
```diff
@@ -42,7 +42,64 @@ class UIController {
...
+                                    <tr>
+                                        <th scope="row">Done</th>
+                                        <td>Shows the % of labeled unique texts over total unique texts.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Wrong Scheme messages</th>
+                                        <td>Shows the number of ‘messages with a label’ that contain a checked “WS (Wrong Scheme)” label.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">WS %</th>
```
Comment (IsaackMwenda) [inline public/UI.js:89]: Include descriptions for %

File: public/UI.js [hunk 10]
```diff
@@ -42,7 +42,64 @@ class UIController {
                     </thead>
                     <tbody id="coding-status-body"></tbody>
                 </table>
-            <div id="last-update">Last updated: </div>
+                <div id="last-update">Last updated: </div>
+                <div class="accordion" id="accordionExample">
+                    <div class="card">
+                        <div id="headingOne">
+                            <h2 class="mb-0">
+                                <button onclick="window.scrollBy(0, 1000)" class="btn btn-brown" type="button" data-toggle="collapse" data-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
```
Comment (IsaackMwenda) [inline public/UI.js]: Thanks for the improvements here. When you load the page the first time, the scroll down does not seem to work. You have to scroll up. It, however, works from then on.PTAL

**Review by IsaackMwenda — CHANGES_REQUESTED** [review]
This LGTM, only requesting changes because I would like to take a final look at the accordion button behaviour

File: public/UI.js (hunk 7 above)
Comment (DanielMwendwa) [reply public/UI.js]: resolved

File: public/UI.js [hunk 11]
```diff
@@ -42,7 +45,64 @@ class UIController {
...
+                                        <th scope="row">WS %</th>
+                                        <td>Percentage of Wrong Scheme messages</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Not Coded messages</th>
+                                        <td>Shows the number of messages with a label’ that contain a checked “NC (Not Coded)” label.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">NC %</th>
+                                        <td>Percentage of Not Coded messages</td>
```
Comment (IsaackMwenda) [inline public/UI.js]: ### 
```suggestion
                                        <td>Shows the percentage of 'messages with a label’ that contain a checked “NC (Not Coded)” label.</td>
```

File: public/UI.js [hunk 12]
```diff
@@ -42,7 +45,64 @@ class UIController {
...
+                                        <th scope="row">Done</th>
+                                        <td>Shows the % of labeled unique texts over total unique texts.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">Wrong Scheme messages</th>
+                                        <td>Shows the number of ‘messages with a label’ that contain a checked “WS (Wrong Scheme)” label.</td>
+                                    </tr>
+                                    <tr>
+                                        <th scope="row">WS %</th>
+                                        <td>Percentage of Wrong Scheme messages</td>
```
Comment (IsaackMwenda) [inline public/UI.js]:
```suggestion
                                        <td>Shows the percentage of ‘messages with a label’ that contain a checked “WS (Wrong Scheme)” label.</td>
```

**Review by IsaackMwenda — APPROVED** [review]

**Pull request merged**